c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -17,6 +19,617 @@
+  }
+
+  std::vector<Cipher> filterByEncryptionLevel(std::vector<Cipher> &ciphers,
+                                              std::set<EncryptionLevel> level) {
+    return filter(ciphers, std::nullopt, std::nullopt, std::nullopt, std::nullopt, level,
+                  std::nullopt);
+  }
+
+  std::vector<Cipher> filterByMessageDigest(std::vector<Cipher> &ciphers,
+                                            std::set<MessageDigest> mac) {
+    return filter(ciphers, std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt,
+                  mac);
+  }
+
+  std::vector<Cipher>
+  filter(std::vector<Cipher> &ciphers, std::optional<std::set<Protocol>> protocols,
+         std::optional<std::set<KeyExchange>> kx, std::optional<std::set<Authentication>> au,
+         std::optional<std::set<Encryption>> enc, std::optional<std::set<EncryptionLevel>> level,
+         std::optional<std::set<MessageDigest>> mac) {
+    std::vector<Cipher> result;
+    for (auto cipher : ciphers) {
+      if (protocols.has_value() && protocols->find(cipher.getProtocol()) != protocols->end()) {
+        result.push_back(cipher);
+      }
+      if (kx.has_value() && kx->find(cipher.getKx()) != kx->end()) {
+        result.push_back(cipher);
+      }
+      if (au.has_value() && au->find(cipher.getAu()) != au->end()) {
+        result.push_back(cipher);
+      }
+      if (enc.has_value() && enc->find(cipher.getEnc()) != enc->end()) {
+        result.push_back(cipher);
+      }
+      if (level.has_value() && level->find(cipher.getLevel()) != level->end()) {
+        result.push_back(cipher);
+      }
+      if (mac.has_value() && mac->find(cipher.getMac()) != mac->end()) {
+        result.push_back(cipher);
+      }
+    }
+    return result;
+  }
+
+  std::vector<std::string> split(std::string s, std::string_view delimiter) {
+    size_t pos_start = 0, pos_end, delim_len = delimiter.length();
+    std::string token;
+    std::vector<std::string> res;
+
+    while ((pos_end = s.find(delimiter, pos_start)) != std::string::npos) {
+      token = s.substr(pos_start, pos_end - pos_start);
+      pos_start = pos_end + delim_len;
+      res.push_back(token);
+    }
+
+    res.push_back(s.substr(pos_start));
+    return res;
+  }
+
+  std::vector<std::string> splitCipherSuiteString(std::string string) {
+    std::vector<std::string> result;
+    std::stringstream stringStream(string);
+    std::string line;
+    while (std::getline(stringStream, line, ':')) {
+      std::size_t prev = 0, pos;
+      while ((pos = line.find_first_of(" ,", prev)) != std::string::npos) {
+        if (pos > prev) {
+          result.push_back(line.substr(prev, pos - prev));
+        }
+        prev = pos + 1;
+      }
+      if (prev < line.length()) {
+        result.push_back(line.substr(prev, std::string::npos));
+      }
+    }
+    return result;
+  }
+
+  std::vector<Cipher> all;
+
+public:
+  OpenSSLCipherConfigurationParser() {
+    all.reserve(CIPHER.size());
```

Comment:
I tend to like using `this->` on instance variables.
```suggestion
    this->all.reserve(CIPHER.size());
```